scene/main/viewport.cpp

```diff
@@ -1563,43 +1567,43 @@ void Viewport::_gui_input_event(Ref<InputEvent> p_event) {
 				set_input_as_handled();
 			}
 
-			if (gui.drag_data.get_type() != Variant::NIL && mb->get_button_index() == MouseButton::LEFT) {
+			if (get_tree()->get_root()->gui.drag_data.get_type() != Variant::NIL && mb->get_button_index() == MouseButton::LEFT) {
```

Comment:
With this PR, `gui.drag_data` is only used on the root Viewport. I am thinking about making it a Viewport static class-variable. This approach would also help with readability, since `get_tree()->get_root()` is no longer needed.